ui/components/schema/EditResolverDialog.js

```diff
@@ -0,0 +1,72 @@
+import BaseResolverDialog from "./BaseResolverDialog";
+
+const INITIAL_STATE = {
+    dataSourceName: null,
+    requestmapping: null,
+    responseMapping: null,
+    err: "",
+    validations: {
+        name: "success",
+        requestmapping: "success",
+        responseMapping: "success"
+    }
+};
+
+class EditResolverDialog extends BaseResolverDialog {
+
+    constructor(props) {
+        super(props);
+        this.state = INITIAL_STATE;
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.dataSource && this.props.dataSource !== prevProps.dataSource) {
```

Comment:
IHMO it could be encapsulated in a function and called here in order to make it more readable. Following an example. 

```javascript
function boolean isPropertiesChanges(prevProps) {
    return this.props.dataSource && this.props.dataSource !== prevProps.dataSource
}

...
if (sPropertiesChanges(prevProps)) {
   {...}
}
```